pkg/arvo/sys/vane/clay.hoon

```diff
@@ -3120,123 +3037,75 @@
         work
       ?>  ?=(%nako p.r.u.rut)
       =/  nako  ;;(nako q.r.u.rut)
-      =/  missing  (missing-blobs nako)
-      =.  let.need.sat  (welp let.need.sat ~(tap in let.missing))
-      =.  old.need.sat  (welp old.need.sat ~(tap in old.missing))
-      =.  lat.ran  (uni-blobs lat.ran hav.missing)
+      =/  missing  (missing-lobes nako)
+      =.  need.sat  (welp need.sat ~(tap in missing))
       =.  nako.sat  (~(put to nako.sat) ~ nako)
       work
     ::
-    ++  missing-blobs
+    ++  missing-lobes
       |=  =nako
-      ^-  [let=(set lobe) old=(set lobe) hav=(map lobe blob)]
-      =/  let-tako  (~(got by gar.nako) let.nako)
-      =|  let-lobes=(set lobe)
-      =|  old-lobes=(set lobe)
-      =|  hav-blobs=(map lobe blob)
+      ^-  (set lobe)
+      =|  missing=(set lobe)
       =/  yakis  ~(tap in lar.nako)
-      |-  ^-  [(set lobe) (set lobe) (map lobe blob)]
+      |-  ^-  (set lobe)
       =*  yaki-loop  $
       ?~  yakis
-        [let-lobes old-lobes hav-blobs]
+        missing
       =/  =norm  (~(gut by tom.dom) r.i.yakis nor.dom)
       =/  lobes=(list [=path =lobe])  ~(tap by q.i.yakis)
-      |-  ^-  [(set lobe) (set lobe) (map lobe blob)]
-      =*  blob-loop  $
+      |-  ^-  (set lobe)
+      =*  lobe-loop  $
       ?~  lobes
         yaki-loop(yakis t.yakis)
+      ?:  (~(has by lat.ran) lobe.i.lobes)
```

Comment:
This might be clearer as a `=?`, to consolidate the conditional and only recurse in one place instead of three.

```hoon
=?    missing
    ?&  !(~(has by lat.ran) lobe.i.lobes)
        !=([~ %|] (~(fit ^de norm) path.i.lobes))
    ==
  (~(put in missing) lobe.i.lobes)
lobe-loop(lobes t.lobes)
```

Reply:
good point